src/imitation/scripts/train_preference_comparisons.py

```diff
@@ -187,20 +191,33 @@ def train_preference_comparisons(
         seed=_seed,
         custom_logger=custom_logger,
     )
+    preference_model = preference_comparisons.PreferenceModel(
+        **cross_entropy_loss_kwargs,
+        model=reward_net,
+    )
+    if active_selection:
+        fragmenter = preference_comparisons.ActiveSelectionFragmenter(
+            preference_model=preference_model,
+            base_fragmenter=fragmenter,
+            fragment_sample_factor=transition_oversampling,
```

Comment:
I have added a new `active_selection_oversampling` variable to set the extra factor of fragment pairs to sample for active selection. 
Just one question: Would `active_selection_oversampling` > `transition_oversampling` make sense? In this case, we are asking for more fragment pairs than available, and so there will be repeated fragments.

Reply:
Hm, I'm a bit rusty on the code, so not sure this is correct (@levmckinney do you want to check my reasoning?) But I think that what's going on is:
  1. `transition_oversampling` controls how many more timesteps we sample from the environment than the number of fragment pairs * fragment length we generate. Even if we had `transition_oversampling < 1` (not usually desirable), we wouldn't necessarily get exact duplicates as we (a) will still sample different pairs, even if we've seen each element in that pair before in another comparison; (b) a trajectory of length M splits into M-N fragments of length N, so as long as fragment length is smaller than trajectory length there's actually a lot of possible (but highly correlated!) fragments.
  2. `active_selection_oversampling` controls how many fragment pairs we sample from the generated fragments for each one we actually label.

So, if we hypothetically did used the `ActiveSelectionFragmenter` but with a totally random active selection criteria, I think this would be the same as just using `RandomFragmenter` but with `transition_oversampling` set to `transition_oversampling / active_oversampling`.

If this is right, then maybe we should multiply `transition_oversampling` by `active_oversampling` to decouple these two hyperparameters?

Reply:
> So, if we hypothetically did used the ActiveSelectionFragmenter but with a totally random active selection criteria, I think this would be the same as just using RandomFragmenter but with transition_oversampling set to transition_oversampling / active_oversampling.

In the case of just using the `RandomFragmenter` with a transition over sample of `transition_oversampling / active_oversampling`. You would have more corolations between the fragments. Apart from that, I agree that the two cases you have outlined behave basically the same.

Let's consider the design decision of how to handle these two hyperparameters in more detail.

## Case 1: We **don't** multiply `tranision_oversampling` by `active_oversampling`.
In this case we calculate how many timesteps to sample from the environment like so,
```python
num_steps = math.ceil(
    self.transition_oversampling * 2 * num_pairs * self.fragment_length,
)
```
Then as a user of the library I can easily control in some sense how “hard we are looking for” surprising pairs in the pool of trajectories we collect. In addition, `active_oversampling` is decoupled from the number of transitions we actually collect from the environment.

## Case 2: We **do** multiply `tranision_oversampling` by `active_oversampling`.
In this case we calculate how many timesteps to sample from the environment like so,
```python
num_steps = math.ceil(
    self.transition_oversampling * self.active_oversampling * 2 * num_pairs * self.fragment_length,
)
```
The amount of correlation among the fragments would be kept to a minimum since we would always sample more from the environment. In addition, the end user might not have to worry about tuning the `self.transition_oversampling` at all. However, now `active_oversampling` also controls how many steps in the environment we are taking which feels a little weird to me.

Thus, I think we should not be multiply them together. We should emphasize that `transition_oversampling` controls how correlated the fragment pairs will be. And `active_oversampling` controls approximately how many of the possible fragment pairs we consider.

Perhaps a better way to decouple these are to replace `active_oversampling` with something like `fraction_of_pairs_to_consider` which controls what fraction of the total set of possible pairs we actually evaluate and rank.